app/protocol/flows/blockrelay/handle_request_ibd_root_utxo_set_and_block.go

```diff
@@ -74,18 +61,38 @@ func (flow *handleRequestIBDRootUTXOSetAndBlockFlow) start() error {
 		}
 
 		// Send the UTXO set in `step`-sized chunks
-		const step = 1024 * 1024 // 1MB
+		const step = 1000
 		offset := 0
 		chunksSent := 0
-		for offset < len(serializedUTXOSet) {
-			var chunk []byte
-			if offset+step < len(serializedUTXOSet) {
-				chunk = serializedUTXOSet[offset : offset+step]
-			} else {
-				chunk = serializedUTXOSet[offset:]
+		for {
+			pruningPointUTXOs, err := flow.Domain().Consensus().GetPruningPointUTXOs(
```

Comment:
Returning an iterator out of `consensus` would break the lock over the latter.
The alternative would be to pass a "start from this outpoint" parameter into `GetPruningPointUTXOs`.
Unfortunately, that would require an additional serialization step within `GetPruningPointUTXOs`.
Maybe we should see first that it's a problem and only then optimize it.